app/apps/server/bridges/livechat.ts

```diff
@@ -87,15 +87,21 @@ export class AppLivechatBridge extends LivechatBridge {
 		return this.orch.getConverters()?.get('rooms').convertRoom(result.room);
 	}
 
-	protected async closeRoom(room: ILivechatRoom, comment: string, appId: string): Promise<boolean> {
+	protected async closeRoom(room: ILivechatRoom, comment: string, closer: IUser | undefined, appId: string): Promise<boolean> {
 		this.orch.debugLog(`The App ${ appId } is closing a livechat room.`);
 
-		return Livechat.closeRoom({
-			visitor: this.orch.getConverters()?.get('visitors').convertAppVisitor(room.visitor),
+		const closeData: any = {
 			room: this.orch.getConverters()?.get('rooms').convertAppRoom(room),
 			comment,
-			user: undefined,
-		});
+		};
+
+		if (closer?.id) {
```

Comment:
You don't need this `if` condition here, you can rely on conditional properties with object spread like the code below:

```
const shouldAddProp2 = false;
const obj = {
  prop1: 2,
  ...shouldAddProp2 && { conditionalProp2: 2 }
}
```
This means that you can implement the logic inside the [closeData](https://github.com/RocketChat/Rocket.Chat/pull/21025/files#diff-0d545a42ed0213be9e61429689f3d0c2012da94d7a7ba988dffd3a02c10b3557R93-R96) object.

Reply:
applied suggestion